velox/exec/NestedLoopJoinBuild.cpp

```diff
@@ -105,6 +105,10 @@ void NestedLoopJoinBuild::noMoreInput() {
     }
   }
 
+  while (dataVectors_.size() > 1) {
+    dataVectors_[0]->append(dataVectors_[dataVectors_.size()-1].get());
```

Comment:
No need to merge the vectors, you just need to switch the looping order during probe to put the build side outmost.